user-modelのテスト

Koh0214 opened the pull request:
# WHAT
user modelのテスト
factory_girlの導入
fakerの導入
binding.pryの導入

# WHY
userモデルが動作することを確認する

Commit:
create pull request

Commit:
rspec-railsのbundle install

Commit:
rails g rspec

Commit:
.rspecに--format documentationを追加

Commit:
userモデルのname, emailのテストコード記述

Commit:
last_commentsのエラーが出たので、gemをアップデートして対応

Commit:
コメント内のnickname => name

Commit:
user modelにvalidateを追加

Commit:
user_rspec.rbの記述を日本語に変更

Koh0214 commented:
![2017-01-11 15 18 14](https://cloud.githubusercontent.com/assets/22772196/21837717/41e2609a-d811-11e6-8ad1-e46b1886d8c5.png)

Koh0214 commented:
![2017-01-11 15 21 01](https://cloud.githubusercontent.com/assets/22772196/21837830/f17a94b4-d811-11e6-915e-c919bfbe7235.png)

Commit:
空行の追加

hc0208 commented on spec/models/user_spec.rb:
```diff
@@ -0,0 +1,21 @@
+    	expect(user.errors[:email]).to include("が入力されていません。")
+    end
+  end
+end
```
describe User do
describe '#create' do
は同じ記述なので1つにまとめて書くことができます。

hc0208 replied:
作成に成功した時のテストも書きましょう。

Koh0214 replied:
まとめました！

成功時のテストコードについて調べます！

Commit:
空行の追加2

Commit:
nameとemailをまとめ、describe #createのネストを削除

Commit:
passwordについてのvalidateを追加

ついでにnameの6文字制限をなくした。

Commit:
factory_girlの導入とディレクトリ、ファイルの作成

Commit:
factory_girlの雛形のusers.rbの中身を編集

Commit:
FactoryGirlの記述を省略するための記述

user = FactoryGirl.build(:user)  =>  user = build(:user)

Commit:
factory_girlを使って書き換え

Commit:
全てのバリデーションをクリアした際のテストコードを追加

Commit:
重複Emailのテストコードを追加

Commit:
パスワードの最小値を8文字に設定

Commit:
テストするものを追加

・重複するemailは登録できない
・8文字以下のパスワードは登録できない
・8文字以上のパスワードは登録できる
・パスワードがあってもpasword_confirmationがないと登録できない

Koh0214 commented:
![2017-01-11 16 44 14](https://cloud.githubusercontent.com/assets/22772196/21839871/82d9c3ca-d81d-11e6-902a-c6ce6dc7b79e.png)

b-i-hide commented on spec/factories/users.rb:
```diff
@@ -0,0 +1,10 @@
+FactoryGirl.define do
+
```
テストデータを作る際にFakerという便利なgemがあるので調べて使ってみましょう！

Koh0214 replied:
調べて使ってみます！

b-i-hide commented on spec/models/user_spec.rb:
```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+describe User do
+  it "is valid with a nickname, email, password, password_confirmation" do
```
テストを書く場合、正常系と異常系でcontextをわけられるとより見やすくなるので、そこもかき分けてみましょう！
今回の場合は、引数が揃っている場合と揃っていない場合などで分けるとより見やすくなります。

Koh0214 replied:
ありがとうございます！

修正します！

Koh0214 replied:
修正しました！
![2017-01-11 17 00 15](https://cloud.githubusercontent.com/assets/22772196/21840200/900a1b7e-d81f-11e6-8b78-fb82a2773562.png)

madana-i replied:
itだけではなくcontextというブロックがあるので、そちらを調べてみて下さい！

Commit:
valid, invalidを見やすくするように入れ替え

Commit:
fakerの導入

Commit:
Fakerで置き換え

passwordとpassword_confirmationは同じものにするため変数passを定義

Commit:
binding.pryの導入

factoryで使われるnameやemail、passがどうなっているのかを確かめた。

Koh0214 commented:
![2017-01-11 17 56 17](https://cloud.githubusercontent.com/assets/22772196/21841827/5a6cc48c-d827-11e6-8143-e4f0003b4631.png)

madana-i commented on Gemfile:
```diff
@@ -35,6 +35,10 @@ gem 'sdoc', '~> 0.4.0', group: :doc
   gem 'byebug'
+  gem 'rspec-rails'
+  gem 'factory_girl_rails'
+  gem 'faker'
```
上記3つのgemはtest環境のみにインストールするようにしてもいいと思います！

madana-i commented on spec/factories/users.rb:
```diff
@@ -0,0 +1,12 @@
+FactoryGirl.define do
+
+pass = Faker::Internet.password(8)
```
ここもブロックの中なのでインデント下げましょう

Koh0214 replied:
修正しました！

Commit:
インデントの修正

Commit:
contextを使い、valid、invalidで分け、invalidの中をさらに細分化した

Koh0214 commented:
![2017-01-11 18 20 23](https://cloud.githubusercontent.com/assets/22772196/21842638/0bf58088-d82b-11e6-8eae-02f358347ebf.png)

Commit:
development, testどちらもに入っていたtest用のgemをtestのみの方へ移した

bundle installで入れたgemのアンインストールの方法

ターミナルで
bundle exec gem uninstall ‘gem名’

Gemfile.lockを削除(bundle installとかすれば生成される)

Gemfileの中の消したgemの記述を削除
